src/components/Badge.js

```diff
@@ -1,13 +1,28 @@
 import React from 'react';
 import PropTypes from 'prop-types';
-import {omit} from 'ramda';
 import {Badge as RSBadge} from 'reactstrap';
+import Link from '../private/Link';
 
 const Badge = props => {
-  const {children, loading_state, ...otherProps} = props;
+  const {children, href, loading_state, setProps, ...otherProps} = props;
+
+  const incrementClicks = () => {
+    if (setProps) {
+      setProps({
+        n_clicks: props.n_clicks + 1,
+        n_clicks_timestamp: Date.now()
+      });
+    }
+  };
+
+  const useLink = href != null;
+  otherProps[useLink ? 'preOnClick' : 'onClick'] = incrementClicks;
```

Comment:
The first line here is just checking if `href` is falsy, so I think it's clearer to not bother with `useLink` and have

```js
otherProps[href ? 'preOnClick' : 'onClick'] = incrementClicks;
```

This is also closer to what is done by e.g. [Button](https://github.com/facultyai/dash-bootstrap-components/blob/master/src/components/Button.js#L24), just using `href` as a truth value.